src/main/java/chatbot/ChatBot.java

```diff
@@ -0,0 +1,70 @@
+package chatbot;
+
+import java.util.Scanner;
+
+import chatbot.action.Action;
+import chatbot.action.ByeAction;
+import chatbot.action.exception.ActionException;
+import chatbot.parse.InputParser;
+import chatbot.storage.LocalStorage;
+import chatbot.task.TaskList;
+import chatbot.ui.Printer;
+
+/**
+ * This encapsulates the behaviour of a chatbot,
+ * which is the handling of the message content and executing commands.
+ *
+ * @author Titus Chew
+ */
+public class ChatBot {
+    /** Stores the scanner instance used to get the console input stream. */
+    private static final Scanner SCANNER = new Scanner(System.in);
+
+    /** Stores the name of this. */
+    private final String chatBotName;
+
+    /** Stores the user's tasks. */
+    private final TaskList userList;
+
+    /**
+     * Class constructor.
+     *
+     * @param chatBotName the name of this chatbot.
+     */
+    public ChatBot(String chatBotName) {
+        this.chatBotName = chatBotName;
+        userList = LocalStorage.loadTaskList();
+    }
+
+    public static void main(String[] args) {
```

Comment:
Can add JavaDocs to this method? I am not too sure about this since the guide only mentioned about overriding and getters/setters being exceptions to this.